pkg/api/v1beta3/dynakube/telemetryservice/props.go

```diff
@@ -0,0 +1,39 @@
+package telemetryservice
+
+type Protocol string
+
+const (
+	OtlpProtocol   Protocol = "otlp"
+	ZipkinProtocol Protocol = "zipkin"
+	JaegerProtocol Protocol = "jaeger"
+	StatsdProtocol Protocol = "statsd"
+)
+
+func KnownProtocols() []Protocol {
+	return []Protocol{
+		OtlpProtocol,
+		ZipkinProtocol,
+		JaegerProtocol,
+		StatsdProtocol,
+	}
+}
+
+func (spec *Spec) GetProtocols() []Protocol {
+	if spec == nil {
+		return []Protocol{}
+	}
+
+	if len(spec.Protocols) == 0 {
+		return KnownProtocols()
+	}
+
+	var protocols []Protocol
+	for _, proto := range spec.Protocols {
+		protocols = append(protocols, Protocol(proto))
+	}
+	return protocols
+}
+
+func (spec *Spec) IsEnabled() bool {
+	return spec != nil
+}
```

Comment:
This will go boom, as the `Spec` is nillable, make it the method of the `TelemetryService` struct (similar to [KSPM](https://github.com/Dynatrace/dynatrace-operator/blob/ef1cf622916eea0875725c83d18a70b4829160c5/pkg/api/v1beta3/dynakube/kspm/props.go#L7-L10))
```
func (ts *TelemetryService) IsEnabled() bool {
	return ts.Spec != nil
}
``` 

you can have `IsEnabled` defined for the `Spec` in case of the ActiveGate as that section is not nillable

Reply:
FYI: so funny thing, it (the original that seems totally impossible) works (Adam tested it, I double checked)
```
func TestBoom(t *testing.T) {
	boom := TelemetryService{
		Spec: nil,
	}

	require.False(t, boom.IsEnabled())
}
```

just to not be confusing, we will still move the method to `TelemetryService`

Reply:
TelemetryServiceSpec.IsEnabled replaced with TelemetryService.IsEnabled